euphonic/readers/phonopy.py

```diff
@@ -543,28 +543,38 @@ def _extract_summary(filename: str, fc_extract: bool = False
         summary_dict['pc_to_sc_atom_idx'] = pc_to_sc_atom_idx
         summary_dict['sc_to_pc_atom_idx'] = sc_to_pc_atom_idx
         summary_dict['ufc'] = pu['force_constants']
-        try:
+
+        if 'force_constants' in summary_object:
             fc = summary_object['force_constants']
             summary_dict['force_constants'] = np.array(
                 fc['elements']).reshape(fc['shape'] + [3,3])
-        except KeyError:
-            pass
+
         # NAC factor may be present even if born is not in phonopy.yaml
         # (born may be in separate BORN file), and the BORN file may
         # not necessarily contain NAC factor, so just try reading it
         # from phonopy.yaml anyway
-        try:
+
+        # New format
+        if 'nac' in summary_object:
+            nac = summary_object['nac']
+            summary_dict['nac_factor'] = nac.get('unit_conversion_factor')
+            summary_dict['born'] = nac.get('born_effective_charge')
+            summary_dict['dielectric'] = nac.get('dielectric_constant')
+
+        # Old format
+        else:
             summary_dict['nac_factor'] = summary_object[
-                'phonopy']['nac_unit_conversion_factor']
-        except KeyError:
-            pass
-        try:
-            summary_dict['born'] = np.array(
-                summary_object['born_effective_charge'])
-            summary_dict['dielectric'] = np.array(
-                summary_object['dielectric_constant'])
-        except KeyError:
-            pass
+                'phonopy'].get('nac_unit_conversion_factor')
+            summary_dict['born'] = summary_object.get('born_effective_charge')
+            summary_dict['dielectric'] = summary_object.get(
+                'dielectric_constant')
+
+        # Cast arrays and drop keys if data not found
+        for key in ('nac_factor', 'born', 'dielectric'):
+            if summary_dict[key] is None:
+                del summary_dict[key]
+            elif key in ('born', 'dielectric'):
+                summary_dict[key] = np.array(summary_dict[key])
```

Comment:
For the sake off saving a loop through 2 elements in a non-performance critical application, I find a second loop more explicit.
```suggestion

        for key in ('born', 'dielectric'):
            summary_dict[key] = np.array(summary_dict[key])
```

Reply:
This will create array(None) if the keys were not present in the file.

Reply:
Ah wait, actually we already deleted it so we will get a KeyError?

Reply:
Adding
```
if key in summary_dict:
```
?
It was a suggestion as I find it describes what's going on better, definitely not required.

Reply:
Like this?

```python
        # Drop NAC keys if no data was found
        for key in ('nac_factor', 'born', 'dielectric'):
            if summary_dict[key] is None:
                del summary_dict[key]

        for key in ('born', 'dielectric'):
            if key in summary_dict:
                summary_dict[key] = np.array(summary_dict[key])
```